src/main/java/org/folio/consortia/service/impl/SyncPrimaryAffiliationServiceImpl.java

```diff
@@ -93,14 +97,24 @@ public void createPrimaryUserAffiliations(UUID consortiumId, SyncPrimaryAffiliat
           var user = userList.get(idx);
           log.info("Processing users: {} of {}", idx + 1, userList.size());
 
-          Page<UserTenantEntity> userTenantPage = userTenantRepository.findByUserId(UUID.fromString(user.getId()), PageRequest.of(0, 1));
-          if (userTenantPage.getTotalElements() > 0) {
-            log.info("Primary affiliation already exists for tenant/user: {}/{}", tenantId, user.getUsername());
-          } else {
-            userTenantService.createPrimaryUserTenantAffiliation(consortiumId, tenantEntity, user.getId(), user.getUsername());
-            sendCreatePrimaryAffiliationEvent(tenantEntity, user);
+          // context changes in every iteration and folioExecutionContext become an empty, so we should set saved context again.
+          try (var context2 = new FolioExecutionContextSetter(prepareContextForTenant(centralTenantId, folioModuleMetadata, currentTenantContext))) {
```

Comment:
We don't need to change to centralTenantId here, for example, we want to retrieve all users from member tenant A, so we don't need switch to central tenant context here, we should remain context from Okapi Headers in this place

Reply:
They are many context changes happening. We have to set context after every context change. One of them is missing, it may cause some exception error. Personally, I don't have any idea to optimize

Reply:
In Future, it may solved. But currently for me it is like **context puzzle**

Reply:
I got it now, context3 after executing closes current context , and context 2 switches context to another tenant, so you need to create context3.